website/docs/topics/feature-flags/best-practices-using-feature-flags-at-scale.md

```diff
@@ -0,0 +1,350 @@
+
+Instrumenting your code in this way has several benefits:
+
+* Simplified Code: By controlling a new feature at a single location, the rest of your system doesn't need to be aware of the flag, leading to cleaner code. 
+* Enhanced Testing: Easier isolation of the feature for unit and integration tests.
+* Improved maintainability: When a feature flag is used at a single location, it is easier to clean up the flag when it has served its purpose. In addition, simpler and testable code is easier to maintain over time.
+
+For backend changes (e.g., a new database schema), evaluating the feature flag near the affected module is best.
+
+This enables:
+
+* Focused Testing: Evaluating the flag's impact within the specified module simplifies testing and debugging. You only need to consider the immediate code's behavior, making troubleshooting more efficient.
+* Isolation: Evaluating close to the module helps isolate the impact of the feature flag, reducing the risk of unintended consequences in other parts of the codebase.
+* Testable: Make sure your modules are testable. Usually, it makes sense to evaluate the flag logic outside and either inject the result or simply have two different implementations of the module. This optimizes for testability, making it straightforward to implement unit tests for both the old and the new logic. 
+
+Once you have defined your flags, striving for single evaluation points whenever possible is crucial. This reduces code complexity and potential errors from managing multiple evaluation points.
+
+Here are a few examples of functionality managed by feature flags and where you should try to evaluate in your code:
+
+* Recommendation Engine: When testing changes to the recommendation algorithm, evaluating the feature flag close to the recommendation module allows focused testing and isolation of the changes.
+* Database Query Optimization: Evaluating the flag near the optimized query simplifies testing and ensures the change behaves as expected within that specific query.
+* Third-Party Integrations: If a feature flag controls the integration with a third-party service, evaluating it near the integration code helps isolate the behavior changes. This simplifies testing the interaction with the third-party service based on the flag state, without affecting other integrations in the system.
+* Cache Invalidation Logic: When modifying code that invalidates cache entries based on a feature flag, evaluating the flag near that logic allows for focused testing of the cache invalidation behavior. This isolates the test from other parts of the system that could potentially interact with the cache.
+
+### 8. Evaluate a feature flag only once.
+
+![evaluate-flags-only-once](https://github.com/ferrantim/unleash/assets/3068350/2c8ffe74-7972-429f-a0be-adcb045a2836)
```

Comment:
I think this image came out pretty nice in the end 😄 

Same thing as for previous images: move it to the docs repo and give it a caption. 

I'm assuming you've called it `/img/evaluate-flags-only-once.png`:

```suggestion
<Figure img="/img/evaluate-flags-only-once.png" caption="You should evaluate flags only once in a chain of services and pass the result on to the next service in the chain if it needs the value. If you evaluate the flag in each service, you risk the flag value changing between each service, causing unexpected results."/>
```